pkg/gofr/datasource/sql/db.go

```diff
@@ -231,37 +230,49 @@ func (d *DB) Select(ctx context.Context, data interface{}, query string, args ..
 
 	switch rv.Kind() {
 	case reflect.Slice:
-		rows, err := d.QueryContext(ctx, query, args...)
-		if err != nil {
-			d.logger.Errorf("error running query: %v", err)
+		d.selectSlice(ctx, query, args, rvo, rv)
 
-			return
-		}
+	case reflect.Struct:
+		d.selectStruct(ctx, query, args, rv)
 
-		for rows.Next() {
-			val := reflect.New(rv.Type().Elem())
+	default:
+		d.logger.Debugf("a pointer to %v was not expected.", rv.Kind().String())
+	}
+}
 
-			if rv.Type().Elem().Kind() == reflect.Struct {
-				d.rowsToStruct(rows, val)
-			} else {
-				_ = rows.Scan(val.Interface())
-			}
+func (d *DB) selectSlice(ctx context.Context, query string, args []interface{}, rvo, rv reflect.Value) {
+	rows, err := d.QueryContext(ctx, query, args...)
+	if err != nil {
+		d.logger.Errorf("error running query: %v", err)
+		return
+	}
 
-			rv = reflect.Append(rv, val.Elem())
-		}
+	for rows.Next() {
+		val := reflect.New(rv.Type().Elem())
 
-		if rvo.Elem().CanSet() {
-			rvo.Elem().Set(rv)
+		if rv.Type().Elem().Kind() == reflect.Struct {
+			d.rowsToStruct(rows, val)
+		} else {
+			_ = rows.Scan(val.Interface())
 		}
 
-	case reflect.Struct:
-		rows, _ := d.QueryContext(ctx, query, args...)
-		for rows.Next() {
-			d.rowsToStruct(rows, rv)
-		}
+		rv = reflect.Append(rv, val.Elem())
+	}
 
-	default:
-		d.logger.Debugf("a pointer to %v was not expected.", rv.Kind().String())
+	if rvo.Elem().CanSet() {
+		rvo.Elem().Set(rv)
+	}
+}
+
+func (d *DB) selectStruct(ctx context.Context, query string, args []interface{}, rv reflect.Value) {
+	rows, err := d.QueryContext(ctx, query, args...)
+	if err != nil {
+		d.logger.Errorf("error running query: %v", err)
+		return
+	}
+
+	for rows.Next() {
+		d.rowsToStruct(rows, rv)
 	}
```

Comment:
No error management to do with something like this?


```suggestion
	for rows.Next() {
		d.rowsToStruct(rows, rv)
 	}

	if err:= rows.Err() {
		return err
 	}
```

Reply:
I have added a check for `rows.Err()` and if it is not nil returning from there after logging